week1/plot_trips.R

```diff
@@ -47,16 +75,33 @@ trips_with_weather <- inner_join(trips, weather, by="ymd")
 
 # plot the number of trips as a function of the minimum temperature, where each point represents a day
 # you'll need to summarize the trips and join to the weather data to do this
-
+by_day <- group_by(trips_with_weather, date) %>% summarize(count = n())
+by_t_and_d <- group_by(trips_with_weather, ymd, tmin) %>% summarize()
+new_col <- select(by_day, count)
+x <- cbind(by_t_and_d, new_col)
+ggplot(data = x, mapping = aes(x = tmin, y = ymd)) + geom_point(aes(size = count))
 # repeat this, splitting results by whether there was substantial precipitation or not
 # you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
-
+trips_with_weather_and_p <- trips_with_weather
+trips_with_weather_and_p$sub_prcp <- with(trips_with_weather, ifelse(prcp > 1.50, "T", "F"))
+a <- group_by(trips_with_weather_and_p, ymd, sub_prcp) %>% summarize(count = n())
+new_col <- select(a, sub_prcp)
+new_x <- cbind(x, new_col)
+ggplot(data = new_x, mapping = aes(x = tmin, y = ymd...1)) + geom_point(aes(size = count, color = sub_prcp))
 # add a smoothed fit on top of the previous plot, using geom_smooth
-
+ggplot(data = new_x, mapping = aes(x = tmin, y = ymd...1)) + geom_point(aes(size = count, color = sub_prcp)) + geom_smooth()
 # compute the average number of trips and standard deviation in number of trips by hour of the day
 # hint: use the hour() function from the lubridate package
-
+trips_hour <- mutate(trips, hour = hour(starttime))
+trips_by_hour <- trips_hour %>% group_by(ymd, hour) %>% summarize(count = n())
+hour_total <- group_by(trips_by_hour, hour) %>% summarize(avg = mean(count), std_dev = sd(count))
 # plot the above
-
+ggplot(data = hour_total) + geom_line(mapping = aes(x = hour, y = avg)) + geom_ribbon(mapping = aes(x = hour, ymin = avg - std_dev, ymax = avg + std_dev), alpha = 0.2)
 # repeat this, but now split the results by day of the week (Monday, Tuesday, ...) or weekday vs. weekend days
 # hint: use the wday() function from the lubridate package
+trips_hour <- mutate(trips_hour, day = wday(ymd))
+trips_hour <- mutate(trips_hour, day=factor(day, levels=c(1,2,3,4,5,6,7), labels=c("Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday")))
+trips_by_day <- trips_hour %>% group_by(ymd, day) %>% summarize(count = n())
+day_total <- group_by(trips_by_day, day) %>% summarize(avg = mean(count), std_dev = sd(count))
+ggplot(data = day_total) + geom_point(mapping = aes(x = day, y = avg))
+ggplot(data = day_total) + geom_point(mapping = aes(x = day, y = std_dev))
```

Comment:
This probably should've been made clear since other students did this too, but we really wanted a series of plots like the previous one, with a seperate plot for each day, that shows both average and std over the course of the day.